File: queue/Queue.java
package queue;

import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Queue {

    private static List<Integer> rotate(List<Integer> list) {
        ArrayDeque<Integer> deque = new ArrayDeque<>(list);
        while (deque.peek() != 0) {
            deque.add(deque.poll());
        }
        return new ArrayList<Integer>(deque);
    }

    private static List<String> splitText(String s) {
        return s.isEmpty() ? List.of() : Arrays.asList(s.split(" "));
    }

    public static void main(String[] args) {
        try (Scanner scan = new Scanner(System.in)) {
            List<Integer> nums = splitText(scan.nextLine()).stream().map(Integer::parseInt)
                    .collect(Collectors.toList());
            var result = rotate(nums);
            System.out.println("Result : [ " + result.stream().map(String::valueOf).collect(Collectors.joining(" "))+ " ]");
        }
    }
}

File: queue/queue.py
#!/usr/bin/env python
from collections import deque

def rotate(nums : list[int]) -> list[int]:
    q = deque(nums)
    while q[0] != 0:
        q.append(q.popleft())
    return list(q)    


if __name__ == "__main__":
    nums = [int(x) for x in input().split()]
    res = rotate(nums)
    print("Result : ")
    print(" ".join(map(str,res)))